frontend/src/metabase/query_builder/components/filters/FilterPopover/FilterPopover.unit.spec.tsx

```diff
@@ -154,4 +129,46 @@ describe("FilterPopover", () => {
       });
     });
   });
+  describe("filter rendering", () => {
+    describe("no-value filters: %s filter", () => {
+      it.each(["is-null", "not-null", "is-empty", "not-empty"])(
+        "should not render picker or separator when selecting a new filter from the column dropdown",
+        async operator => {
+          setup({
+            filter: [operator, ["field", PRODUCTS.TITLE, null], null] as Filter,
+            showFieldPicker: false,
+          });
+
+          expect(
+            screen.queryByTestId("filter-popover-separator"),
+          ).not.toBeInTheDocument();
+
+          // if the "empty filter picker" boolean check fails, we will see the default filter picker
+          expect(
+            screen.queryByTestId("default-filter-picker"),
+          ).not.toBeInTheDocument();
+        },
+      );
+    });
+
+    describe("non datetime filters", () => {
+      it.each([
+        { filterElem: STRING_CONTAINS_FILTER, label: "contains" },
+        { filterElem: NUMERIC_FILTER, label: "equals" },
+      ])(
+        "should render the default filter picker with a separator if the $label filter has arguments",
+        async ({ filterElem }) => {
+          setup({ filter: filterElem });
```

Comment:
Two different names refer to the same thing both here and in `setup`. We should choose one.

```suggestion
        { filter: STRING_CONTAINS_FILTER, label: "contains" },
        { filter: NUMERIC_FILTER, label: "equals" },
      ])(
        "should render the default filter picker with a separator if the $label filter has arguments",
        async ({ filter }) => {
          setup({ filter });
```